library/tiqr/tests/Tiqr_StateStoragePdoTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+require_once 'tiqr_autoloader.inc';
+
+use Mockery as m;
+use Mockery\MockInterface;
+use PHPUnit\Framework\TestCase;
+
+class Tiqr_StateStoragePdoTest extends TestCase
+{
+    /**
+     * @var Tiqr_StateStorage_Pdo_TestDouble
+     */
+    private $stateStorage;
+
+    /**
+     * @var MockInterface|PDO
+     */
+    private $pdoHandle;
+
+    private function makeTempDir() {
+        $tempName = tempnam(sys_get_temp_dir(),'Tiqr_StateStorageTest');
+        unlink($tempName);
+        mkdir($tempName);
+        return $tempName;
+    }
+
+    private function setUpDatabase(string $dsn)
+    {
+        // Create test database
+        $pdo = new PDO($dsn, null, null, [PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION]);
+        $pdo->exec("CREATE TABLE state (key varchar(255) PRIMARY KEY, expire int, value text);");
+    }
+
+    protected function setUp(): void
+    {
+        $targetPath = $this->makeTempDir();
+        $dsn = 'sqlite:' . $targetPath . '/state.sq3';
+        $this->setUpDatabase($dsn);
+
+        $options = array(
+            'table' => 'state',
+            'dsn' => $dsn,
+            'username' => null,
+            'password' => null,
+        );
+
+        $pdoHandle = new PDO($options['dsn'],$options['username'],$options['password']);
+        $this->pdoHandle = m::spy($pdoHandle)->makePartial();
+        $this->stateStorage = new Tiqr_StateStorage_PdoTestDouble($options, $pdoHandle);
+        $this->assertInstanceOf(Tiqr_StateStorage_Pdo::class, $this->stateStorage);
+    }
+
+    function test_called_clean_expired() {
+        $key = random_bytes(3);
+        $this->stateStorage->setValue($key, random_bytes(5), 1);
+
+        $value = $this->stateStorage->getValue($key);
+        $this->stateStorage->shouldHaveReceived('cleanExpired');
```

Comment:
Effective way to test that cleanup was called. Nice!

Reply:
Wait till you see the version that actually worked ;)